src/helpers/component-tree.ts

```diff
@@ -87,3 +87,24 @@ export function getHostSiblings(
     (sibling) => !hostSelves.includes(sibling)
   );
 }
+
+export function getCompositeParentOfType(
+  element: ReactTestInstance,
+  type: React.ComponentType
+) {
+  let current = element.parent;
+
+  while (!isHostElement(current)) {
+    // We're at the root of the tree
+    if (!current) {
+      return null;
+    }
+
+    if (current.type === type) {
+      return current;
+    }
+    current = current.parent ?? null;
```

Comment:
Is that null coalescing necessary? According to @types/react then type of parent is `ReactTestInstance | null`
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2b7c710f4890784cd537c82bdcd2c03e98ec09db/types/react-test-renderer/index.d.ts#L31